File: Loiane_Training/Java_Intermediario/Exer Aula 53 - 56/Calculadora/Calculadora.java
package Calculadora;

public class Calculadora {
    public static void main(String[] args) {
        double soma = Operacoes.SOMAR.executarOperacao(10, 10);
        double subtracao = Operacoes.SUBTRAIR.executarOperacao(10, 10);
        double multiplicacao = Operacoes.MULTIPLICAR.executarOperacao(10, 10);
        double divisao = Operacoes.DIVIDIR.executarOperacao(10, 10);

        System.out.println("Soma: " + soma);
        System.out.println("Subtracao: " + subtracao);
        System.out.println("Multiplicacao: " + multiplicacao);
        System.out.println("Divisao: " + divisao);
    }
}

enum Operacoes {
    SOMAR("+") {
        @Override
        public double executarOperacao(double x, double y) {
            return x + y;
        }
    },
    SUBTRAIR("-") {
        @Override
        public double executarOperacao(double x, double y) {
            return x - y;
        }
    },
    MULTIPLICAR("*") {
        @Override
        public double executarOperacao(double x, double y) {
            return x * y;
        }
    },
    DIVIDIR("/") {
        @Override
        public double executarOperacao(double x, double y) {
            return x / y;
        }
    };

    Operacoes(String operacao) {
        this.operacao = operacao;
    }

    private String operacao;

    public String getOperacao() {
        return this.operacao;
    }

    public abstract double executarOperacao(double x, double y);
}

File: Loiane_Training_C#/Basico/I/Ex3/Calculadora.cs
using System;

class Calculadora
{
    static double Somar(double x, double y)
    {
        return x + y;
    }

    static double Subtrair(double x, double y)
    {
        return x - y;
    }

    static double Multiplicar(double x, double y)
    {
        return x * y;
    }

    static double Dividir(double x, double y)
    {
        return x / y;
    }

    static double Elevacao(double _base, double expoente)
    {
        return Math.Pow(_base, expoente);
    }

    static int CalcularFatorial(int x)
    {
        int fatorial = 1;
        for (int i = x; i > 0; i--)
        {
            fatorial *= i;
        }

        return fatorial;
    }

    static void Main(string[] args)
    {
        double soma = Calculadora.Somar(5, 5);
        Console.WriteLine("Soma: " + soma);

        double subtracao = Calculadora.Subtrair(5, 5);
        Console.WriteLine("Subtracao: " + subtracao);

        double multiplicacao = Calculadora.Multiplicar(5, 5);
        Console.WriteLine("Multiplicacao: " + multiplicacao);

        double divisao = Calculadora.Dividir(5, 5);
        Console.WriteLine("Divisao: " + divisao);

        double potenciacao = Calculadora.Elevacao(5, 5);
        Console.WriteLine("Potenciacao: " + potenciacao);

        int fatorial = Calculadora.CalcularFatorial(6);
        Console.WriteLine("Fatorial: {0}", fatorial);
    }
}